src/Core/AdminConsole/OrganizationFeatures/OrganizationUsers/ImportOrganizationUserCommand.cs

```diff
@@ -0,0 +1,428 @@
+    /// <param name="importUserData">Data containing existing and imported external users along with their Id mappings.</param>
+    private async Task OverwriteExisting(
+            List<(OrganizationUserUserDetails ou, EventType e, DateTime? d)> events,
+            OrganizationUserImportData importUserData)
+    {
+        var usersToDelete = importUserData.ExistingExternalUsers.Where(u =>
+            u.Type != OrganizationUserType.Owner &&
+            !importUserData.ImportedExternalIds.Contains(u.ExternalId) &&
+            importUserData.ExistingExternalUsersIdDict.ContainsKey(u.ExternalId));
+        await _organizationUserRepository.DeleteManyAsync(usersToDelete.Select(u => u.Id));
+        events.AddRange(usersToDelete.Select(u => (
+          u,
+          EventType.OrganizationUser_Removed,
+          (DateTime?)DateTime.UtcNow
+          ))
+        );
+        foreach (var deletedUser in usersToDelete)
+        {
+            importUserData.ExistingExternalUsersIdDict.Remove(deletedUser.ExternalId);
+        }
+    }
+
+    /// <summary>
+    /// Imports group data into the organization by saving new groups and updating existing ones.
+    /// </summary>
+    /// <param name="organization">The organization into which groups are being imported.</param>
+    /// <param name="importedGroups">A collection of groups to be imported.</param>
+    /// <param name="importUserData">Data containing information about existing and imported users.</param>
+    private async Task ImportGroups(Organization organization,
+            IEnumerable<ImportedGroup> importedGroups,
+            OrganizationUserImportData importUserData)
+    {
+
+        if (!importedGroups.Any())
+        {
+            return;
+        }
+
+        if (!organization.UseGroups)
+        {
+            throw new BadRequestException("Organization cannot use importedGroups.");
+        }
+
+        var existingGroups = await _groupRepository.GetManyByOrganizationIdAsync(organization.Id);
+        var importGroupData = new OrganizationGroupImportData(importedGroups, existingGroups);
+
+        await SaveNewGroups(importGroupData, importUserData);
+        await UpdateExistingGroups(importGroupData, importUserData, organization);
+    }
+
+    /// <summary>
+    /// Saves newly imported groups that do not already exist in the organization.
+    /// Sets their creation and revision dates, associates users with each group.
+    /// </summary>
+    /// <param name="importGroupData">Data containing both imported and existing groups.</param>
+    /// <param name="importUserData">Data containing information about existing and imported users.</param>
+    private async Task SaveNewGroups(OrganizationGroupImportData importGroupData, OrganizationUserImportData importUserData)
+    {
+        var newGroups = importGroupData.Groups
+            .Where(g => !importGroupData.ExistingExternalGroups.ToDictionary(g => g.ExternalId!).ContainsKey(g.Group.ExternalId!))
```

Comment:
I assume that this will call `ToDictionary` every time it executes this lambda for each item in `importGroupData.Groups`. Maybe linq's smart enough to avoid this? But I assume not. Best to construct the dictionary outside of the callback.